src/dump.c

```diff
@@ -2810,7 +2810,7 @@ static void jl_recache_other(void)
             _new = (jl_value_t*)jl_recache_method_instance((jl_method_instance_t*)o, i);
         }
         else {
-            assert(0);
+            abort();
```

Comment:
This should not be worse than what was there before. I can put a message if anyone has any suggestions.